tfx/dsl/component/experimental/function_parser.py

```diff
@@ -60,6 +59,17 @@ class ArgFormats(enum.Enum):
 _OPTIONAL_PRIMITIVE_MAP = dict((Optional[t], t) for t in _PRIMITIVE_TO_ARTIFACT)
 
 
+def _get_return_type_annotations(typehints: Dict[str, Any]) -> Optional[Dict]:
+  """Returns annotations of expected return types."""
+  return_annotations = typehints.get('return')
+  if hasattr(return_annotations, "__annotations__"):
+    return return_annotations.__annotations__
```

Comment:
Either typing.get_type_hints (if you want resolved type) or inspect.get_annotations() is (if you want a raw __annotations__) should be preferred.